auraed/src/runtime/cells/cell.rs

```diff
@@ -82,73 +80,83 @@ impl Cell {
 
     pub fn start_executable(
         &mut self,
-        exe_name: ExecutableName,
-        command: Command,
+        executable_name: ExecutableName,
+        command: String,
         args: Vec<String>,
         description: String,
     ) -> Result<()> {
+        // TODO: replace with try_insert when it becomes stable
         // Check if there was already an executable with the same name.
-        if self.executables.contains_key(&exe_name) {
+        if self.executables.contains_key(&executable_name) {
             return Err(CellError::ExecutableExists {
                 cell_name: self.name.clone(),
-                executable_name: exe_name,
+                executable_name,
             });
         }
 
+        let executable =
+            Executable::new(executable_name, command, args, description);
+        let executable_name = executable.name.clone();
+
+        // Ignoring return value as we've already assured ourselves that the key does not exist.
+        let _ = self.executables.insert(executable_name.clone(), executable);
+
         // Start the child process
-        let exe =
-            Executable::start(exe_name.clone(), command, args, description)
-                .map_err(|e| CellError::ExecutableError {
+        if let Some(executable) = self.executables.get_mut(&executable_name) {
```

Comment:
we could not ignore the return value of insert and save a lookup here, no?

Reply:
ie, i don't know what we'd put in an "else" here so this feels a bit redundant.

Reply:
Since we insert it into the `HashMap` on the right above, we've lost ownership of `executable`. Inside the `if` we have a `&mut` to executable. 

The lookup is technically fallible, but we know it won't fail because we just called `insert` and there is a `MutexGuard` on the entire `CellsTable`, so nothing can happen in between. 

The else would be `unreachable!("executable is guaranteed to be in the HashMap")`, which would panic if the assumptions of it definitely being there is wrong. I left that out, but it is probably safer to put that in.

Reply:
Added the `unreachable`